### ISSUE TITLE
Add support for "mdc-menu-surface--fixed" in ui-select

### ISSUE THREAD

**Issue by leibale** [balmjs/balm-ui#24]
Hi,

In order to use the select component inside of an element that has `overflow: hidden` without the droplist being cut we need to add the "mdc-menu-surface--fixed" class to the ".mdc-select__menu" element (https://github.com/material-components/material-components-web/issues/5590#issuecomment-657032119)

An argument like `fullwidth` will be very appreciated :)

Thanks

**Comment by elf-mouse** [comment]
Hi @leibale ,  Thank you so much for this issue :) 

I am preparing for the release of a new version this week.
And I will check and fix this bug in the new version.

**Comment by elf-mouse** [comment]
Hi @leibale , 

This bug is fixed now!

- `balm-ui@8.7.1` for Vue 2
- `balm-ui@9.0.1` for Vue 3

You can use new prop `fixedWidth` like:

```html
<ui-dialog v-model="open">
  <ui-dialog-content>
    <ui-select :options="options" :fixed-width="200"></ui-select>
  </ui-dialog-content>
  <ui-dialog-actions></ui-dialog-actions>
</ui-dialog>
```

**Comment by jfrag** [comment]
Hi,

Get the same issue with ui-autocomplete on a dialog.

**Comment by elf-mouse** [comment]
@leibale ,

It's improved.

- `balm-ui@8.7.3` for vue2
- `balm-ui@9.1.0` for vue3

**Comment by elf-mouse** [comment]
@jfrag ,

`<ui-autocomplelte>` has no such problems in the `<ui-dialog>`, because the dialog is scrollable.

**Comment by leibale** [comment]
@elf-mouse I think that something went worng in the release of last version, cause I updated to `9.1.0` and I can't find the `fixedWidth` attribute on `<ui-select>` (I've also opened `node_modules/balm-ui/src/scripts/components/selection-controls/select.vue`- the `fixedWidth` attribute is not there...)

**Comment by elf-mouse** [comment]
Yes, after internal discussions in our team, we felt that this prop was not very user-friendly, so we restored this prop in `9.0.2`.

Current interaction effect:

- Default dialog
![Default](https://user-images.githubusercontent.com/2597430/100035472-3a3caa80-2e39-11eb-9c6a-7ada60177512.png)
- Select in dialog
![Select](https://user-images.githubusercontent.com/2597430/100035477-3dd03180-2e39-11eb-9013-af37e7a14d2f.png)
- Autocomplete in dialog
![Autocomplete](https://user-images.githubusercontent.com/2597430/100035479-3f99f500-2e39-11eb-882a-110db1333bdd.png)

**Comment by leibale** [comment]
@elf-mouse so how can I solve the issue in a custom component that has `overflow` set to `hidden`?

**Comment by elf-mouse** [comment]
Hi @leibale , 

It's improved again in `9.2.0`, the select is auto fixed in dialog now.

Set an explicit width by `data-width`

```html
<ui-dialog>
  <ui-dialog-content>
    <ui-select data-width="200px"></ui-select>
  </ui-dialog-content>
</ui-dialog>
```

![overflow inside of dialog](https://user-images.githubusercontent.com/2597430/100199178-ce8d3700-2f37-11eb-858e-7c28150864d7.png)

**Comment by jfrag** [comment]
> @jfrag ,
> 
> `<ui-autocomplelte>` has no such problems in the `<ui-dialog>`, because the dialog is scrollable.

ok. it's not user friendly...
it's possible to have a parameter for ui-autocomplete to overflow the dialog

**Comment by elf-mouse** [comment]
@jfrag ,

Yes. but `<ui-select>` and `<ui-autocomplete>` are implemented differently.
I will consider improving this issue for `<ui-autocomplete>` in the next version.

Please new issue for `<ui-autocomplete>`.

Thanks again :)

**Comment by jfrag** [comment]
Ok

Thank you for your help and your good work !

**Comment by leibale** [comment]
@elf-mouse is there a way to enable it only for a specific `<ui-select>` and set the width on the element?

**Comment by elf-mouse** [comment]
Hi @leibale , `balm-ui@9.3.0` released :)

Now, you can use `<ui-select fixed data-width="xxx">` for the select inside a overflow hidden component.

**Comment by leibale** [comment]
@elf-mouse

Currently, the `fixed` boolean is saved in the global `state` variable, which means that there's no way to have 2 `<ui-select>` at the same time on the page while having one of them inside an overflow hidden component and the other one in a regular component (without the `mdc-menu-surface--fixed` class).
Also, I don't understand when `state.fixed` goes back to `false` after it's being set to `true`.

thank you for this amazing module, and your work! :)

**Comment by elf-mouse** [comment]
Hi @leibale ,

This is a small bug, I have fixed it in `9.3.1`.

```js
app.use(BalmUI, {
  UiSelect: {
    fixed: true // I don't recommend the default fixed, but you can set according to the actual scene
  }
});
```

Thank you very much for your long-term support :)
